native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -110,6 +118,11 @@ public static Object toJson(BXml xml, boolean preserveNamespaces, Type type) {
      */
     public static Object convertToJSON(BXml xml, String attributePrefix, boolean preserveNamespaces, Type type,
                                        BMap<BString, BString> parentAttributeMap) throws Exception {
+        if (type != null && type.toString().contains(MAP_XML)) {
```

Comment:
Can't we check the type or some attribute of the type instead of doing a string search?

Reply:
Updated in [dfe5e7d](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/431/commits/dfe5e7dedb5743868cfea01e85d934cd4c7faaa5)